### PR TITLE
[Tests] Refactored parsing of scheme from repo URL, added Collection tests

### PR DESCRIPTION
**Pull request by czechboy0**
- refactored parsing of scheme from repo URL
- moved Array utils to Collection utils 
- added Collection tests 
- moved all allTests functions into one file (Utilities modules only for now), previously they were in FileTests which was hard to find

### CODE REVIEW

File: Sources/Utility/CollectionExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                a.append(e)
+            } else {
+                b.append(e)
+            }
+        }
+        return (a, b)
+    }
+}
+
+//FIXME figure out why it doesn't work when Iterator.Element == Equatable
```
Comment (gribozavr) [inline Sources/Utility/CollectionExtensions.swift]: Use `Iterator.Element : Equatable`

Comment (czechboy0) [inline Sources/Utility/CollectionExtensions.swift]: Ah, thank you!!!

**Comment by mxcl** [comment]
I appreciate the effort, but this wasn't merged earlier because it is really several PRs. The additional tests are most interesting to me, and I'm happy to have a general Collection utils organization. The rest is not really desired, though I certainly can see the improvements, I'd rather improve the tests in general, but I'm waiting on the Linux test generation work.

I understand URL.scheme is not general, but I dislike `repoScheme`, and it's not as though this is public API in Cocoa or anything.

**Comment by czechboy0** [comment]
@mxcl I renamed `repoScheme` back to `scheme`, so those changes are reverted. What's left is the repo scheme parsing refactoring and the new tests and moved manifests. Are those the changes you wanted to keep? If not, please let me know which part exactly you have reservations about.

**Comment by mxcl** [comment]
@swift-ci Please test

**Comment by czechboy0** [comment]
@mxcl @swift-ci seems to be in a lazy mood

**Comment by mxcl** [comment]
@swift-ci Please test

**Comment by czechboy0** [comment]
Hitting the same WatchKit test failure as everyone else.

**Comment by czechboy0** [comment]
@mxcl Please rerun CI

**Comment by mxcl** [comment]
@swift-ci Please test

**Comment by czechboy0** [comment]
\o/

**Pull request merged**